### PR TITLE
[RFC] Cleanup/pylint

### PR DESCRIPTION
**Pull request by dwaynebailey**
I did a number of pylint cleanups and added these to permanent checks.

* New checks: uneeded-not, unused-import, unused-variable
* Enabled checking in migrations to get some added cleanups
* With `unused-variables` I took a naming decision that such vars are either `__` (double underscore) or `sensible_variable_name_` (where you still want to convey the variable name but indicate that it is unused with a trailing underscore). 

Would love feedback on what is proposed here.

### CODE REVIEW

**Comment by phlax** [comment]
>  unused-import, unused-variable

These would already be picked up by pyflakes i figure, but no harm to add

> Enabled checking in migrations to get some added cleanups

Not really sure if there is any point to this, as most migrations are created and run automatically. The other migrations are `do once and forget` so there may be some benefit to linting, but it would be pretty limited

> With unused-variables I took a naming decision that such vars are either __

Looks like a good plan

File: pootle/apps/pootle_app/management/commands/update_tmserver.py [hunk 1]
```diff
@@ -352,5 +352,4 @@ def handle(self, **options):
         if self.is_local_tm:
             self._set_latest_indexed_revision(**options)
 
-        success, _ = helpers.bulk(self.es,
-                                  self._parse_translations(**options))
+        __, __ = helpers.bulk(self.es, self._parse_translations(**options))
```
Comment (unho) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: In this case it might make more sense to just remove the assignment.

File: pootle/apps/pootle_fs/resources.py [hunk 2]
```diff
@@ -163,7 +163,7 @@ def pootle_changed(self):
 
     def reload(self):
         """Uncache cached_properties"""
-        for k, v in self.__dict__.items():
+        for k, __ in self.__dict__.items():
```
Comment (unho) [inline pootle/apps/pootle_fs/resources.py]: Why not `v_` following what you did in other places?

File: pootle/apps/pootle_project/models.py [hunk 3]
```diff
@@ -142,7 +142,7 @@ class ProjectURLMixin(object):
     """
 
     def get_absolute_url(self):
-        lang, proj, dir, fn = split_pootle_path(self.pootle_path)
+        __, proj, __, __ = split_pootle_path(self.pootle_path)
```
Comment (unho) [inline pootle/apps/pootle_project/models.py]: I would use here `whatever_` syntax.

File: pootle/apps/pootle_store/migrations/0010_set_store_filetypes.py [hunk 4]
```diff
@@ -16,7 +16,6 @@ def migrate_store_filetypes(apps, schema_editor):
 
     projects = apps.get_model("pootle_project.Project").objects.all()
     stores = apps.get_model("pootle_store.Store").objects.all()
-    filetypes = apps.get_model("pootle_format.Format").objects.all()
```
Comment (unho) [inline pootle/apps/pootle_store/migrations/0010_set_store_filetypes.py]: This belongs to a separate cleanup commit.

File: pootle/apps/pootle_fs/resources.py (hunk 2 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_fs/resources.py]: I've only used `$var_` if I thought it added value to understanding.  I think it's pretty clear here that this is `key`, `value`. And I don't think it helps here.

File: pootle/core/views.py [hunk 5]
```diff
@@ -778,7 +778,7 @@ def get_context_data(self, *args, **kwargs):
             raise Http404(
                 ValidationError(search_form.errors).messages)
 
-        total, start, end, units_qs = search_backend.get(Unit)(
+        total, __, __, units_qs = search_backend.get(Unit)(
```
Comment (unho) [inline pootle/core/views.py]: In cases where several `__` are assigned at once I would keep the `whatever_` syntax.

File: pootle/checks.py [hunk 6]
```diff
@@ -267,11 +267,11 @@ def check_settings(app_configs=None, **kwargs):
...
             try:
                 if markup_filter == 'textile':
-                    import textile  # noqa
+                    import textile  # noqa pylint:disable=unused-variable
                 elif markup_filter == 'markdown':
-                    import markdown  # noqa
+                    import markdown  # noqa pylint:disable=unused-variable
                 elif markup_filter == 'restructuredtext':
-                    import docutils  # noqa
+                    import docutils  # noqa pylint:disable=unused-variable
```
Comment (unho) [inline pootle/checks.py]: This unused variables cleanups belong to another commit.

File: pootle/core/markup/filters.py [hunk 7]
```diff
@@ -72,11 +72,11 @@ def get_markup_filter():
...
             return (None, "unset")
         elif markup_filter == 'textile':
-            import textile  # noqa
+            import textile  # noqa pylint:disable=unused-variable
         elif markup_filter == 'markdown':
-            import markdown  # noqa
+            import markdown  # noqa pylint:disable=unused-variable
         elif markup_filter == 'restructuredtext':
-            import docutils  # noqa
+            import docutils  # noqa pylint:disable=unused-variable
```
Comment (unho) [inline pootle/core/markup/filters.py]: And these unused variables cleanups belong to another commit too.

File: pootle/apps/pootle_project/models.py (hunk 3 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_project/models.py]: I took the decision that `split_pootle_path` is pretty well understood and thus `lang_` clouds the issue and is not useful.  What this does show though is that we probably want helper functions that actually return `proj` from `self.pootle_path`

**Comment by unho** [comment]
lgtm overall, only some nits. Just wondering about Travis, but just restarted the job to see if it is actually picking something.

File: pootle/core/views.py (hunk 5 above)
Comment (dwaynebailey) [inline pootle/core/views.py]: My rationale was simple.  `__` makes it much more readable then `something_` so I'm aiming for readability.  The only reason to keep vars is if the function creating them is not that well known and you want to keep that in the code.

File: pootle/apps/pootle_fs/resources.py (hunk 2 above)
Comment (phlax) [inline pootle/apps/pootle_fs/resources.py]: in this case we dont need the placeholder at all, just remove the `.items()`

File: pootle/apps/pootle_project/models.py (hunk 3 above)
Comment (phlax) [inline pootle/apps/pootle_project/models.py]: how about:

```
proj  = split_pootle_path(self.pootle_path)[1]
```

File: pootle/checks.py (hunk 6 above)
Comment (phlax) [inline pootle/checks.py]: no!

i defo dont agree to *any* linting that requires disabling comments

Comment (phlax) [inline pootle/checks.py]: also, this is highlighting bad coding - please dont disable it

Comment (dwaynebailey) [inline pootle/checks.py]: We already do.  I'm also anti, except where the places we need it are minor and the overall benefit is larger enough.  I.e. where we have no false positives other that these ones which we can tightly control and where we're operating slightly out of normal.

Of course suggestions to tighten the code are the other option.

Comment (phlax) [inline pootle/checks.py]: these are not false positives

Comment (phlax) [inline pootle/checks.py]: in fact if anything we should remove the `#noqa` and fix, not hide again

Comment (dwaynebailey) [inline pootle/checks.py]: Open to suggestions on how to fix the errors that will present here.

**Comment by dwaynebailey** [comment]
Had a quick chat with @phlax to discuss use of `pylint` comments.  Our general conclusion:

1. These are usually bad bits of code and they shouldn't be hidden
2. Land the fixes that the check picked up
3. Flag a cleanup bug and highlight that the check can be enabled if fixed
4. Don't enable check by default till its properly fixed, especially if we have alternatives in other linters that are doing similar things without reporting an issue.

**Comment by julen** [comment]
>> Enabled checking in migrations to get some added cleanups
> 
> Not really sure if there is any point to this, as most migrations are created and run automatically. 

I agree, I don't believe there's much point on this either.

File: pootle/apps/pootle_fs/resources.py (hunk 2 above)
Comment (unho) [inline pootle/apps/pootle_fs/resources.py]: :+1:

File: pootle/checks.py (hunk 6 above)
Comment (phlax) [inline pootle/checks.py]: @dwaynebailey i reckon we could fix this relatively easily with importlib

heres a SO discussion about similar http://stackoverflow.com/questions/14050281/how-to-check-if-a-python-module-exists-without-importing-it

File: pytest_pootle/utils.py [hunk 8]
```diff
@@ -132,7 +132,7 @@ def get_translated_storefile(store, pootle_path=None):
     """Returns file store with added translations for untranslated units."""
     storeclass = store.get_file_class()
     filestore = store.convert(storeclass)
-    for i, unit in enumerate(filestore.units):
+    for i_, unit in enumerate(filestore.units):
```
Comment (phlax) [inline pytest_pootle/utils.py]: if the `i` is not used we can safely remove the enumeration

File: tests/misc/checks.py [hunk 9]
```diff
@@ -402,7 +402,7 @@ def test_get_qualitycheck_schema():
             'url': ''
         })
 
-    result = sorted([item for code, item in d.items()],
+    result = sorted([item for code_, item in d.items()],
                     key=lambda x: x['code'],
```
Comment (phlax) [inline tests/misc/checks.py]: we can use `d.values()` and remove `code_`

File: tests/views/admin.py [hunk 10]
```diff
@@ -249,7 +249,7 @@ def test_admin_view_projects(client, request_users, english):
     languages = Language.objects.exclude(code='templates')
     language_choices = [(lang.id, unicode(lang)) for lang in languages]
     filetypes = []
-    for name, info in formats.get().items():
+    for name_, info in formats.get().items():
```
Comment (phlax) [inline tests/views/admin.py]: we can use `formats.get().values()` here to get rid of `name_`
